aclk/agent_cloud_link.c

```diff
@@ -40,11 +50,11 @@ int cloud_to_agent_parse(JSON_ENTRY *e)
                 break;
             }
             if (!strcmp(e->name, ACLK_JSON_IN_TOPIC)) {
-                data->topic = strdupz(e->data.string);
+                data->callback_topic = strdupz(e->data.string);
                 break;
             }
             if (!strcmp(e->name, ACLK_JSON_IN_URL)) {
-                data->url = strdupz(e->data.string);
+                data->payload = strdupz(e->data.string);
                 break;
             }
             break;
```

Comment:
No reason for atol just copy paste from other part of the agent code. 

Actually there is a bug here. When the field is number the  `data.string` is not available. Other parts of the code use original_string. So it will change to atoi along with that bug fix.

Reply:
Considering that there was not a specific motive, I will give suggestion, `atoi`, `atol` and `atoll` are functions that call `strtol`, case it is possible, call direct this last casting the result ;).

Reply:
I already put atoi -- but good point :+1: